src/runtime/manifest.ts

```diff
@@ -1364,25 +1364,15 @@ ${e.message}
 
   // TODO: This is a temporary method to allow sharing stores with other Arcs.
   registerStore(store: UnifiedStore, tags: string[]): void {
-    if (!this.findStoreById(store.id) && tags.includes('shared')) {
+    // Only register stores that have non-volatile storage key and don't have a
+    // #volatile tag.
+    if (!this.findStoreById(store.id) &&
+        !store.storageKey.toString().startsWith('volatile')
```

Comment:
don't use toString().startsWith to test NG keys - look directly at the protocol instead. It does mean you need to have 2 code paths in the short term.

Reply:
done.